components/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog/src/main/webapp/WEB-INF/beans.xml

```diff
@@ -63,6 +63,7 @@
   <!-- Out interceptors -->
   <bean id="gZipInterceptor" class="org.apache.cxf.transport.common.gzip.GZIPOutInterceptor" />
   <bean id="responseInterceptor" class="org.wso2.carbon.apimgt.rest.api.util.interceptors.response.ResponseOutInterceptor" />
+  <bean id="cleanUpInterceptor" class="org.wso2.carbon.apimgt.rest.api.util.interceptors.response.CleanUpInterceptor" />
```

Comment:
Do we need to add this for other apis also?

Reply:
@malinthaprasan aiya, since we clear threads in fault flow and response flow (of the publisher API), do we need to add them to other APIs?

Reply:
From where do we set the thread Local? Is this through authentication interceptor?
if its through authentication interceptor, we need to add the cleanup interceptor into all the apis where authentication interceptor is added